File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        s_count = defaultdict(int)
        t_count = defaultdict(int)
        
        for i in range(len(s)):
            s_count[s[i]] += 1
            t_count[t[i]] += 1
            
        return s_count == t_count
            
            
        

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> hm1 = new HashMap<Character, Integer>();
        HashMap<Character, Integer> hm2 = new HashMap<Character, Integer>();

        for (char c: s.toCharArray()) {
            hm1.put(c, hm1.getOrDefault(c, 0) + 1);
        }
        for (char c: t.toCharArray()) {
            hm2.put(c, hm2.getOrDefault(c, 0) + 1);
        }

        return hm1.equals(hm2);


    }
}